File: rehovot-ads-packages/android/src/main/java/com/rehovottech/levelplay/Logger.kt
package com.rehovottech.levelplay

import android.util.Log

// Small logging wrapper so debug output can be enabled without touching business logic.
class Logger(
    private val tag: String,
    private var debugEnabled: Boolean = false,
) {
    public fun setDebugEnabled(enabled: Boolean) {
        debugEnabled = enabled
    }

    public fun d(message: String) {
        if (debugEnabled) {
            Log.d(tag, message)
        }
    }

    public fun i(message: String) {
        if (debugEnabled) {
            Log.i(tag, message)
        }
    }

    public fun w(message: String) {
        if (debugEnabled) {
            Log.w(tag, message)
        }
    }

    public fun e(message: String, throwable: Throwable? = null) {
        if (debugEnabled) {
            Log.e(tag, message, throwable)
        }
    }
}

File: rehovot-ads-packages/ios/Plugin/Logger.swift
import Foundation

// Lightweight logger keeps debug output controllable without coupling to SDK logic.
final class Logger {
    private let tag: String
    private var debugEnabled: Bool

    init(tag: String, debugEnabled: Bool = false) {
        self.tag = tag
        self.debugEnabled = debugEnabled
    }

    func setDebugEnabled(_ enabled: Bool) {
        debugEnabled = enabled
    }

    func debug(_ message: String) {
        guard debugEnabled else { return }
        print("[\(tag)] \(message)")
    }

    func info(_ message: String) {
        guard debugEnabled else { return }
        print("[\(tag)] \(message)")
    }

    func warning(_ message: String) {
        guard debugEnabled else { return }
        print("[\(tag)] \(message)")
    }

    func error(_ message: String) {
        guard debugEnabled else { return }
        print("[\(tag)] \(message)")
    }
}
